### PR TITLE
[FEATURE] introduce RootSiteProcessor

### PR DESCRIPTION
**Pull request by twoldanski**
- add default SiteProvider (combines data from yaml files & database) with optional custom sorting
- add default SiteSchema (outputs domain list for site switcher)
- add default DomainSchema (outputs domain configuration for nuxt-typo3 app)

All providers & schemas are switchable by implementing proper interface
and configure it in data processor options

### CODE REVIEW

File: Classes/DataProcessing/RootSiteProcessing/DomainSchema.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public function process(SiteProviderInterface $provider, array $options = []): array
+    {
+        $processorConfiguration = $options['processorConfiguration'] ?? [];
+        $cObj = $options['cObj'] ?? GeneralUtility::makeInstance(ContentObjectRenderer::class);
+        $result = [];
+
+        foreach ($provider->getSites() as $site) {
+            $protocol = $site->getBase()->getScheme() . '://';
+            $baseUrl = $protocol . $site->getBase()->getHost();
+            $url = $this->siteService->getFrontendUrl($baseUrl, $site->getRootPageId());
```
Comment (lukaszuznanski) [inline Classes/DataProcessing/RootSiteProcessing/DomainSchema.php:56]: are we sure that frontend url is set? it's enabled only if feature switch is turned on

File: Configuration/TypoScript/Configuration/Domains.typoscript [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            replace = 1
+        }
+    }
+
+    10 = JSON
+    10 {
+        fields {
+            domains = TEXT
+            domains {
+                dataProcessing {
```
Comment (lukaszuznanski) [inline Configuration/TypoScript/Configuration/Domains.typoscript:22]: maybe try to run dataProcessing without domains = TEXT? dataProcessing is available for JSON content object without using additional content objects

File: Classes/DataProcessing/RootSiteProcessing/DomainSchema.php (hunk 1 above)
Comment (twoldanski) [reply Classes/DataProcessing/RootSiteProcessing/DomainSchema.php:56]: added check in new commit

File: Configuration/TypoScript/Configuration/Domains.typoscript (hunk 2 above)
Comment (twoldanski) [reply Configuration/TypoScript/Configuration/Domains.typoscript]: Simplified version in new commit

**Review by lukaszuznanski — APPROVED** [review]

**Pull request merged**